### PR TITLE
[DOCS#2082] Documents configurations for CNI policy-setup timeouts

### PR DESCRIPTION
**Pull request by aaaaaaaalex**
Product Version(s):
OSS 3.28 (EE 3.19 EP2 to come at a later date)

Issue:
https://tigera.atlassian.net/browse/DOCS-2082

Link to docs preview:
https://deploy-preview-1373--calico-docs-preview-next.netlify.app/calico/next/reference/configure-cni-plugins#enabling-policy-setup-timeout

https://deploy-preview-1373--calico-docs-preview-next.netlify.app/calico/next/reference/felix/configuration  (See manifest section -> `EndpointStatusPathPrefix` field)

https://deploy-preview-1373--calico-docs-preview-next.netlify.app/calico/next/reference/resources/felixconfig similar to previous link

SME review:
- [ ] An SME has approved this change. 

DOCS review:
- [ ] A member of the docs team has approved this change.

Additional information:

Merge checklist:

- [ ] Deploy preview inspected wherever changes were made 
- [ ] Build completed successfully
- [ ] Test have passed

### CODE REVIEW

File: calico/reference/configure-cni-plugins.mdx [hunk 1]
```diff
@@ -248,6 +248,35 @@ you must also run calico/kube-controllers with the policy, profile, and workload
 
 When using `type: k8s`, the {{prodname}} CNI plugin requires read-only Kubernetes API access to the `Pods` resource in all namespaces.
 
+### Enabling Policy Setup Timeout
+
+If you wish for the CNI to delay a new pod from starting until the pod's policy has been programmed to the dataplane,
```
Comment (ctauchen) [inline calico/reference/configure-cni-plugins.mdx]: we can ignore this Vale error for now.

File: calico/reference/configure-cni-plugins.mdx [hunk 2]
```diff
@@ -248,6 +248,35 @@ you must also run calico/kube-controllers with the policy, profile, and workload
...
+    "policy_setup_timeout_seconds": 10
+  }
+}
+```
+
+:::note
+
+A `policy_setup_timeout_seconds` of 0 disables any policy-setup waiting.
+
+:::
```
Comment (ctauchen) [inline calico/reference/configure-cni-plugins.mdx]: I'm not sure what this note adds that isn't clear from the idea of setting it to 0 seconds. Am I missing something? Can we get rid of this?

File: calico/reference/configure-cni-plugins.mdx [hunk 3]
```diff
@@ -248,6 +248,35 @@ you must also run calico/kube-controllers with the policy, profile, and workload
...
+
+:::note
+
+If the pod's policy is not programmed after the number of seconds specified, the CNI will allow the pod to start its containers anyway.
+
+:::
```
Comment (ctauchen) [inline calico/reference/configure-cni-plugins.mdx]: Let's move this to the intro text, without the note formatting. (I've added to comment above.)

File: calico/reference/configure-cni-plugins.mdx [hunk 4]
```diff
@@ -248,6 +248,35 @@ you must also run calico/kube-controllers with the policy, profile, and workload
 
 When using `type: k8s`, the {{prodname}} CNI plugin requires read-only Kubernetes API access to the `Pods` resource in all namespaces.
 
+### Enabling Policy Setup Timeout
```
Comment (ctauchen) [inline calico/reference/configure-cni-plugins.mdx]:
```suggestion
### Enabling policy setup timeout
```
* We use sentence case for title and headings.

I'm not 100% clear on the distinction in this document between generic and kubernetes-specific configuration. Just want to check this is in the right place. Is the idea that this doesn't apply, for example, to an OpenStack deployment?

File: calico/reference/configure-cni-plugins.mdx (hunk 3 above)
Comment (ctauchen) [reply calico/reference/configure-cni-plugins.mdx]: I think with different phrasing of the intro text, this idea is clear. Let me know what you think.

**Review by ctauchen — COMMENTED** [review]
Thanks for this @aaaaaaaalex. A few comments and suggestions for you.

File: calico/reference/configure-cni-plugins.mdx (hunk 1 above)
Comment (ctauchen) [inline calico/reference/configure-cni-plugins.mdx]: Some new copy, based partly on the previous sections and incorporating the idea of a note you have below:

```
The `policy_setup_timeout_seconds` option makes the CNI plugin wait to start a new pod until one of the following conditions occurs:

* The pod's policy has finished being programmed.
* A specified amount of time has elapsed.

By setting this option, you can avoid errors that can occur when a pod tries to start before the pod's policy is programmed.

Example CNI config:
```

File: calico/reference/configure-cni-plugins.mdx (hunk 2 above)
Comment (aaaaaaaalex) [reply calico/reference/configure-cni-plugins.mdx]: Not missing anything, over verbose  on my part :smile:

File: calico/reference/configure-cni-plugins.mdx (hunk 4 above)
Comment (aaaaaaaalex) [reply calico/reference/configure-cni-plugins.mdx]: Correct - this functionality lives in the k8s portion of our CNI :+1:

File: calico/reference/configure-cni-plugins.mdx (hunk 1 above)
Comment (aaaaaaaalex) [reply calico/reference/configure-cni-plugins.mdx]: Brilliant, thanks!

**Comment by ctauchen** [comment]
LGTM! Ready to merge?

**Comment by aaaaaaaalex** [comment]
Thanks @ctauchen , yes I think we're good to merge :rocket:

**Pull request merged**